src/script/interpreter.cpp

```diff
@@ -965,6 +968,41 @@ bool EvalScript(vector<vector<unsigned char> > &stack,
                 }
                 break;
 
+                case OP_DATASIGVERIFY:
+                {
+                    if (!enableDataSigVerify)
+                        return set_error(serror, SCRIPT_ERR_VERIFY);
+                    if (stack.size() < 3)
+                        return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
+
+                    valtype &data = stacktop(-3);
+                    valtype &vchSigAndType = stacktop(-2);
+                    valtype &vchAddr = stacktop(-1);
+
+                    if (vchAddr.size() != 20)
+                        return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
+
```

Comment:
Add
```
if (vchSigAndType.size() != 66)
    return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
```
?

Reply:
Its a possible addition.  But a sig that's the wrong length won't verify anyway so the check is a little bit redundant.  
Also, the purpose of the type byte is so we can have different sig types in the future that would be different sizes, but of course we could remove or change this line when we added that.  I'm OK with your suggestion since it would limit the attack surface into the signature validation code.

Reply:
You need to be sure that vchSigAndType[0] exists at least

Reply:
added exactly as you recommended